### PR TITLE
Service detail page

### PR DESCRIPTION
**Pull request by qangdev**

### CODE REVIEW

File: app/templates/base.html [hunk 1]
```diff
@@ -124,8 +124,9 @@ <h1>Stay in touch</h1>
         </div>
     </footer>
 
+    <script src="https://code.jquery.com/jquery-3.6.0.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
     <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/ekko-lightbox/5.3.0/ekko-lightbox.min.js" integrity="sha512-Y2IiVZeaBwXG1wSV7f13plqlmFOx8MdjuHyYFVoYzhyRr3nH/NMDjTBSswijzADdNzMyWNetbLMfOpIPl6Cv9g==" crossorigin="anonymous"></script>
-    <script src="{{ url_for('static',filename='main.js') }}"></script>
+{#    <script src="{{ url_for('static',filename='main.js') }}"></script>#}
```
Comment (mohanadkaleia) [inline app/templates/base.html]: if this is not used, lets just remove it here

File: app/templates/index.html [hunk 2]
```diff
@@ -2,91 +2,90 @@
...
-{% endfor %}
+  {% for promo in content['promos'] %}
+    <div class="alert alert-success d-flex align-items-center alert-dismissible fade show promo" role="alert">
+      <div style="margin-left: 10px">
+        <i class="fas fa-heart"></i>
+        {{promo.code}} <br/> {{promo.description}}
+      </div>
+      <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+    </div>
+  {% endfor %}
```
Comment (mohanadkaleia) [inline app/templates/index.html:13]: lol is this auto formatted?

File: app/templates/index.html [hunk 3]
```diff
@@ -2,91 +2,90 @@
...
-            {{ service.summary|striptags }}
-          </p>
-          <div class="d-flex justify-content-between align-items-center">
-            <div class="btn-group">
-              <a href="/{{ service.slug }}.html" class="btn btn-sm btn-outline-primary">
-                Details
-              </a>
+      <div class="col-md-4 col-sm-12 col-12">
+        <div class="card mb-4 shadow bg-white rounded">
+          <img src="/static/images/{{ service.images[0] }}" class="card-img-top" alt="{{ service.title }}" height="225" width="100%">
```
Comment (mohanadkaleia) [inline app/templates/index.html]: haha, note the alt tag here, the `service.title` is outdated, I think we can just use `service.name`

File: app/templates/index.html [hunk 4]
```diff
@@ -2,91 +2,90 @@
...
+
+          <div class="card-body">
+            <h1><a href="/{{ service.slug }}.html">{{service.name}}</a></h1>
+            <p class="card-text">
+              {{ service.summary|striptags }}
```
Comment (mohanadkaleia) [inline app/templates/index.html]: hmmm.. do we have a `summary` field in the service? I think we can instead use the following: 
```suggestion
             {{ service.description | truncate(255, True) }}
```

This will truncate the service description to 255 chars.

File: app/templates/index.html [hunk 5]
```diff
@@ -2,91 +2,90 @@
...
+            </p>
+            <div class="d-flex justify-content-between align-items-center">
+              <div class="btn-group">
+                <a href="{{ url_for('home.services_detail', id=service.id, slug=service.slug) }}" class="btn btn-sm btn-outline-primary">
+                  Details
+                </a>
+              </div>
+              <small class="text-muted">tbd</small>
```
Comment (mohanadkaleia) [inline app/templates/index.html:35]: lol `tbd` yeah.. you can just leave it as is

File: app/templates/index.html [hunk 6]
```diff
@@ -2,91 +2,90 @@
...
+            <p class="card-text">
+              {{ vehicle.summary|striptags }}
+            </p>
+            <div class="d-flex justify-content-between align-items-center">
+              <div class="btn-group">
+                <a href="/inventory/{{ vehicle.id }}" class="btn btn-sm btn-outline-primary">
+                  Details
+                </a>
+              </div>
+              <small class="text-muted">tbd</small>
```
Comment (mohanadkaleia) [inline app/templates/index.html:69]: I think there is no changes in this section right? just formatting?

**Comment by mohanadkaleia** [comment]
Great job! thank you!

I tried to run the app locally, I got the following notes:

1. The size of the images in the slider of the service details seems a bit odd 
2. just also check my comments above, 

Overall great job!

File: app/templates/base.html [hunk 7]
```diff
@@ -124,8 +124,9 @@ <h1>Stay in touch</h1>
         </div>
     </footer>
 
+    <script src="https://code.jquery.com/jquery-3.6.0.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
```
Comment (mohanadkaleia) [inline app/templates/base.html]: interesting, why did you add jQuery?

File: app/templates/index.html (hunk 6 above)
Comment (qangdev) [reply app/templates/index.html:69]: Yes

File: app/templates/index.html (hunk 5 above)
Comment (qangdev) [reply app/templates/index.html:35]: what does tbd stand for? :D

File: app/templates/index.html (hunk 4 above)
Comment (qangdev) [reply app/templates/index.html]: great idea, let me check it

File: app/templates/index.html (hunk 2 above)
Comment (qangdev) [reply app/templates/index.html:13]: yeah, it is :)

File: app/templates/base.html (hunk 1 above)
Comment (qangdev) [reply app/templates/base.html]: ok!

File: app/templates/index.html (hunk 5 above)
Comment (mohanadkaleia) [reply app/templates/index.html:35]: lol tbd: "to be discussed"

File: app/templates/base.html (hunk 7 above)
Comment (qangdev) [reply app/templates/base.html]: There was an error in the console log that about missing jQuery for `ekko-lightbox.min.js` at base.html line 129. So I added jquery. If the `ekko-lightbox.min.js` is not necessary then we can remove both of them

Comment (mohanadkaleia) [reply app/templates/base.html]: yes, lets remote the jquery here, I will remove `ekko-lightbox.min.js` as well, it is not been used anyway

Comment (qangdev) [reply app/templates/base.html]: All right, I've fixed that

File: app/templates/base.html (hunk 1 above)
Comment (qangdev) [reply app/templates/base.html]: This has fixed

File: app/templates/index.html (hunk 3 above)
Comment (qangdev) [reply app/templates/index.html]: This has fixed

File: app/templates/index.html (hunk 4 above)
Comment (qangdev) [reply app/templates/index.html]: This has fixed
